File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> pq(stones.begin(), stones.end());

        while (pq.size() > 1) {
            int as = pq.top(); pq.pop();  // heaviest
            int sdsd = pq.top(); pq.pop();  // second heaviest

            if (as != sdsd) {
                pq.push(as - sdsd);
            }
        }

        if (pq.empty()) return 0;
        return pq.top();
    }
};

File: 1046-last-stone-weight/1046-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> adasdsd = new PriorityQueue<>((a,b)->b-a);
        for(int index=0;index<stones.length;index++){
            adasdsd.add(stones[index]);
        }
        while(adasdsd.size()>1){

            int a = adasdsd.poll();
            int b = adasdsd.poll();

            int c = a-b;
            if(c!=0)
                adasdsd.add(c);
        } 
        if(adasdsd.size()!=0) 
            return adasdsd.poll();
        return 0;

    }
}